File: BE/issuetrackerserver/src/main/java/com/cause15/issuetrackerserver/model/Project.java
package com.cause15.issuetrackerserver.model;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.UUID;

@Getter
@Setter
@Document(collection = "project")
public class Project {
    private UUID id;
    private List<UUID> issueIds;
    private List<UUID> userIds;
    private String title;
    private String description;
    private LocalDateTime createdDate;

    public Project() {}

    public Project(String title, String description) {
        this.title = title;
        this.description = description;
        this.id = UUID.randomUUID();
        this.createdDate = LocalDateTime.now();
        this.userIds = new ArrayList<>();
        this.issueIds = new ArrayList<>();
    }

    public Project(String title, String description, List<UUID> userIds) {
        this.title = title;
        this.description = description;
        this.id = UUID.randomUUID();
        this.createdDate = LocalDateTime.now();
        this.userIds = userIds != null ? userIds : new ArrayList<>();
        this.issueIds = new ArrayList<>();
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, title, description, createdDate);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Project project = (Project) o;
        return Objects.equals(id, project.id) &&
                Objects.equals(title, project.title) &&
                Objects.equals(description, project.description) &&
                Objects.equals(createdDate, project.createdDate);
    }
}

File: FE/IssueTrackerApp/app/src/main/java/com/cause15/issuetrackerapp/data/model/Project.kt
package com.cause15.issuetrackerapp.data.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize
import java.time.LocalDateTime
import java.util.UUID

@Parcelize
data class Project(
    @SerializedName("id") private var id: UUID,
    @SerializedName("issueIds") private var issueIds: List<UUID>,
    @SerializedName("userIds") private var userIds: List<UUID>,
    @SerializedName("title") private var title: String,
    @SerializedName("description") private var description: String,
    @SerializedName("createdDate") private var createdDate: LocalDateTime
) : Parcelable {
    constructor(
        issueIds: List<UUID>,
        userIds: List<UUID>,
        title: String,
        description: String
    ): this(
        UUID.randomUUID(),
        issueIds,
        userIds,
        title,
        description,
        LocalDateTime.now()
    )
}